src/config.js

```diff
@@ -606,36 +646,56 @@ class Config {
   // Returns an {Array} of {Object}s with the following keys:
   //  * `scopeDescriptor` The {ScopeDescriptor} with which the value is associated
   //  * `value` The value for the key-path
-  getAll (keyPath, options) {
+  getAll (keyPath, options = {}) {
     let globalValue, result, scope
-    if (options != null) { ({scope} = options) }
-
-    if (scope != null) {
-      let legacyScopeDescriptor
-      const scopeDescriptor = ScopeDescriptor.fromObject(scope)
-      result = this.scopedSettingsStore.getAll(
-          scopeDescriptor.getScopeChain(),
-          keyPath,
-          options
-        )
-      legacyScopeDescriptor = this.getLegacyScopeDescriptor(scopeDescriptor)
-      if (legacyScopeDescriptor) {
-        result.push(...Array.from(this.scopedSettingsStore.getAll(
-            legacyScopeDescriptor.getScopeChain(),
+    if (options != null) {
+      ({scope} = options)
+    }
+
+    const allResults = []
+    const settings = [this.dirtySettings, ...this.pathSettingsMap.values(), this.projectSettings, this.globalSettings]
+
+    for (let setting of settings) {
+      result = []
+      if (scope != null) {
+        let legacyScopeDescriptor
+        const scopeDescriptor = ScopeDescriptor.fromObject(scope)
+        result = result.concat(setting.scopedSettings.getAll(
+            scopeDescriptor.getScopeChain(),
             keyPath,
             options
-          ) || []))
+          ))
+        legacyScopeDescriptor = this.getLegacyScopeDescriptor(scopeDescriptor)
+        if (legacyScopeDescriptor) {
+          result.push(...Array.from(setting.scopedSettings.getAll(
```

Comment:
why a concat above and a push with spread here?

given the complexity of this and its similarity to the concat call above, I would probably extract this as a function and give it a meaningful name

Reply:
This is actually from the original Atom code - I can refactor this though!